src/pybind/mgr/dashboard/services/access_control.py

```diff
@@ -88,47 +88,63 @@ def check_password_characters(self):
         return self.complexity_credits
 
     def check_is_old_password(self):
+        if not Settings.PWD_POLICY_CHECK_OLDPWD_ENABLED:
+            return False
         return self.old_password and self.password == self.old_password
 
     def check_if_contains_username(self):
+        if not Settings.PWD_POLICY_CHECK_USERNAME_ENABLED:
+            return False
         if not self.username:
             return False
         return self._check_if_contains_word(self.password, self.username)
 
     def check_if_contains_forbidden_words(self):
+        if not Settings.PWD_POLICY_CHECK_EXCLUSION_LIST_ENABLED:
+            return False
         return self._check_if_contains_word(self.password,
                                             '|'.join(self.forbidden_words))
 
     def check_if_sequential_characters(self):
+        if not Settings.PWD_POLICY_CHECK_SEQUENTIAL_CHARS_ENABLED:
+            return False
         for i in range(1, len(self.password) - 1):
             if ord(self.password[i - 1]) + 1 == ord(self.password[i])\
                == ord(self.password[i + 1]) - 1:
                 return True
         return False
 
-    def check_if_repetetive_characters(self):
+    def check_if_repetitive_characters(self):
+        if not Settings.PWD_POLICY_CHECK_REPETITIVE_CHARS_ENABLED:
+            return False
         for i in range(1, len(self.password) - 1):
             if self.password[i - 1] == self.password[i] == self.password[i + 1]:
                 return True
         return False
 
-    def check_password_length(self, min_length=8):
-        return len(self.password) >= min_length
+    def check_password_length(self):
+        if not Settings.PWD_POLICY_CHECK_LENGTH_ENABLED:
+            return True
+        return len(self.password) >= Settings.PWD_POLICY_MIN_LENGTH
 
     def check_all(self):
         """
         Perform all password policy checks.
         :raise PasswordPolicyException: If a password policy check fails.
         """
-        if self.check_password_characters() < 10 or not self.check_password_length():
+        if not Settings.PWD_POLICY_ENABLED:
+            return
+        if self.check_password_complexity() < Settings.PWD_POLICY_MIN_COMPLEXITY:
+            raise PasswordPolicyException('Password is too weak.')
+        if not self.check_password_length():
```

Comment:
I would suggest all the `check_*` methods return the same value if they are valid.
p.e.: `check_is_old_password` returns `false` if valid and `check_password_length` return `true` if valid.

Reply:
Your right with that. I decided against refactoring this to keep the number of changes small. Will think about it again.

Reply:
OK, we can create a issue for further improvements. 

Just found another needed improvement that should be added to that issue:
![image](https://user-images.githubusercontent.com/399326/72803627-14091580-3c3e-11ea-90ca-ea2b2e65ce6e.png)
I have no idea what keyword is invalid, we should display the list of keywords or the exact one.

Reply:
Fixed.
![Bildschirmfoto vom 2020-01-21 16-58-13](https://user-images.githubusercontent.com/1897962/72820446-42531900-3c6f-11ea-9f94-11e0fbd313f2.png)